src/osd/PG.cc

```diff
@@ -3752,11 +3753,18 @@ int PG::build_scrub_map_chunk(
     return ret;
   }
 
-
-  get_pgbackend()->be_scan_list(map, ls, deep, seed, handle);
+  get_pgbackend()->be_scan_list(map, ls, seed, handle);
   _scan_rollback_obs(rollback_obs, handle);
   _scan_snaps(map);
 
+  if (deep) {
+    // XXX: Drop the PG lock because we are just reading the data and 
```

Comment:
@athanatos Doesn't the requeue actually drop the PG lock between each group of N objects?  Even though this code drops the lock after it re-aquires  it, nothing else is done but packaging up and sending the response!